Utilities/convertFromDepsCloud.js

```diff
@@ -0,0 +1,46 @@
+const fs = require('fs')
+      path = require('path')
+
+const depsInputPath =
+    process.env.MARINER_DEPSCLOUD_INPUT ||
+    path.join(__dirname, '..', 'InputFiles', 'candidate.json');
+const depsOutputPath =
+    process.env.MARINER_DEPSCLOUD_OUTPUT ||
+    path.join(__dirname, '..', 'OutputFiles', 'candidate-converted.json');
+
+function convert(deps) {
+  result = {};
+  for(var item in deps) {
+    // Format may change?
+    var first = deps[item].repository_url;
+    var second = deps[item].score;
+    // Assuming all URIs would have a double slash, we follow the slashes to get owner and repo
+    firstpull = first.split("//")[1];
+    cut = firstpull.split("/")[2]
+    if (cut.includes(".git")) {
+      cut = cut.split(".")[0]
+    }
+    first = firstpull.split("/")[1] + "/" + cut;
+    console.log("** ITEM: " + first);
+    result[first] = second;
```

Comment:
Instead of manually parsing, I think you should be able to use Node's `URL` component to parse the URI: https://nodejs.org/api/url.html#url_url_strings_and_url_objects

Reply:
I tried the Node URL component with the current version and I think I actually made it longer 😓 I think this is because the parsing to get owner, to get repo, and to remove the possible `.git` extension from URLs all takes place after extracting the URL pathname. Also open to alternatives. I did find [a parser](https://github.com/IonicaBizau/git-url-parse) that can be used for git URLs

Included is my code while trying Node URLs (I know, not very elegant ;)
```
    const myURL = new URL(deps[item].repository_url)
    cut = myURL.pathname
// Need to extract the owner from a specific spot (Seems to be same place on all git URLs)
    owner = cut.split("/")[1]
// Need to find precise location for the repo
    repo = cut.split("/")[2]
// .git will occasionally show up
    if (repo.includes(".git")) {
      repo = repo.split(".")[0]
    }
// Need to combine to place these in JSON string
    first = owner + "/" + repo;
    console.log("** ITEM: " + first);
    var second = deps[item].score;
    result[first] = second;
```

Reply:
Even if the code is longer, I would favor using node's URL parser. It more clearly conveys intent, and it would handle any legal but unexpected URL formatting that might come along. 

If .git is always at the end (if present), using endsWith would be safer than includes, and you could .replace('.git', ''). Or perhaps even replace '/\.git$' with '', but I haven't tested that, so verify it. 

If the pathname always starts with the same text, you could also use a .replace to remove it. And at that point, could you avoid split entirely? 

I would encourage you to use meaningful naming throughout. So pathname instead of cut, fullRepoIdentifier (or whatever makes sense to you) instead of first, and score instead of second.

Reply:
Thank you, I'm learning a lot from this. I'm going to go through and add some more meaningful names throughout.

Reply:
@kevindigo 

This question in more detail:

> If the pathname always starts with the same text, you could also use a .replace to remove it. And at that point, could you avoid split entirely?

I was able to avoid split after taking your advice, except in one spot:
After obtaining the full path with the `url.pathname` function, I've got a string to work with that goes like so:
```
/owner/repo/other_sections
OR
/owner/repo
```
I haven't found any `pathname` cases that aren't built like these. I treat the string with this code to get the owner and repo:
```
var owner = fullPath.split("/")[1];
var repo = fullPath.split("/")[2];
```
That is the only spot that `split` is still being used since my last commit.

Reply:
Moving discussion to https://github.com/indeedeng/Mariner-Issue-Collector/pull/14#discussion_r508659334